src/components/scene/canvas.js

```diff
@@ -28,6 +28,8 @@ module.exports.Component = register('canvas', {
       canvas.classList.add('a-canvas');
       canvas.style.height = data.height + '%';
       canvas.style.width = data.width + '%';
+      // Mark canvas as provided/injected by A-Frame.
+      canvas.dataset.aframeProvided = true;
```

Comment:
I would say we also add the `aframeProvided` data attribute to the lights/camera (https://github.com/aframevr/aframe/issues/798), but need to keep the `default-light`/`default-camera` so they can be removed individually